node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -1098,17 +1098,55 @@ pub async fn data_request_report(params: Result<(Hash,), jsonrpc_core::Error>) -
         .await
 }
 
+/// Params of getBlockChain method
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct GetBalanceParams {
+    /// Public key hash
+    #[serde(default)] // default to default pkh
+    pub pkh: PublicKeyHash,
+    /// Distinguish between fetching a simple balance or fetching confirmed and unconfirmed balance
+    #[serde(default)] // default to false
+    pub simple: bool,
+}
+
 /// Get balance
-pub async fn get_balance(params: Result<(PublicKeyHash,), jsonrpc_core::Error>) -> JsonRpcResult {
-    let pkh = match params {
-        Ok(x) => x.0,
-        Err(e) => return Err(e),
+pub async fn get_balance(params: Params) -> JsonRpcResult {
+    let (pkh, simple): (PublicKeyHash, bool);
+
+    // Handle parameters as an array with a first obligatory PublicKeyHash field plus an optional bool field
+    if let Params::Array(params) = params {
+        if let Some(Value::String(public_key)) = params.get(0) {
+            match public_key.parse() {
+                Ok(public_key) => pkh = public_key,
+                Err(e) => {
+                    return Err(internal_error(e));
+                }
+            }
+        } else {
+            return Err(jsonrpc_core::Error::invalid_params(
+                "First argument of `get_balance` must have type `PublicKeyHash`",
+            ));
+        };
+
+        match params.get(1) {
+            None => simple = false,
+            Some(Value::Bool(ith)) => simple = *ith,
+            Some(_) => {
+                return Err(jsonrpc_core::Error::invalid_params(
+                    "Second argument of `get_balance` must have type `Bool`",
+                ))
+            }
+        };
+    } else {
+        return Err(jsonrpc_core::Error::invalid_params(
+            "Params of `get_balance` method must have type `Array`",
+        ));
```

Comment:
And here we should parse params as `GetBalanceParams`, otherwise I think in this state the CLI method doesn't work:

```suggestion
        let params: GetBalanceParams = params.parse()?;
        pkh = params.pkh;
        simple = params.simple;
```